lucene/core/src/java/org/apache/lucene/search/WANDScorer.java

```diff
@@ -447,6 +461,23 @@ private int doNextCompetitiveCandidate() throws IOException {
       }
     }
 
+      // nocommit Do we still need the following given TwoPhaseIterator.matches already performs the check
+//     minCompetitiveScore satisfied, now checks if the doc has enough required number of matches
```

Comment:
Indeed I don't think we need it, but I like doing it so that the approximation doesn't return documents knowing that no match is possible.

Reply:
Makes sense.